### PR TITLE
Update evaluator to Scala 2.13.1

### PR DESCRIPTION
**Pull request by kiroco12**
This PR updates the evaluator to Scala 2.13.1. This PR is a Draft because we need to fix a memory leak which cause server crashes in heroku. Libraries updated. Updated Scala's compiler reporters.

### CODE REVIEW

File: server/src/main/scala/org/scalaexercises/evaluator/services.scala [hunk 1]
```diff
@@ -116,12 +115,14 @@ object EvaluatorServer extends IOApp {
   lazy val port = (Option(System.getenv("PORT")) orElse
     Option(System.getProperty("http.port"))).map(_.toInt).getOrElse(8080)
 
+  val evaluator = new Evaluator[IO](10.seconds)
```
Comment (juanpedromoreno) [inline server/src/main/scala/org/scalaexercises/evaluator/services.scala]: Can we deliver this in isolation to the current version 0.5.0? We want to fix the memory leak asap

File: project/ProjectPlugin.scala [hunk 2]
```diff
@@ -123,11 +112,11 @@ object ProjectPlugin extends AutoPlugin {
...
       ),
       orgLicenseSetting := ApacheLicense,
-      scalaVersion := "2.12.10",
+      scalaVersion := "2.13.1",
       scalaOrganization := "org.scala-lang",
       javacOptions ++= Seq("-encoding", "UTF-8", "-Xlint:-options"),
-      scalacOptions += "-Ypartial-unification",
-      fork in Test := false,
+      //javaOptions += "-Dscala.classpath.closeZip=true",
+      //fork in Test := false,
```
Comment (juanpedromoreno) [inline project/ProjectPlugin.scala]: Could we remove this commented code?

File: server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala [hunk 3]
```diff
@@ -264,7 +257,7 @@ private class StringCompiler(
     // ...and 1/2 this line:
     compiler.compileSources(sourceFiles)
 
-    if (reporter.hasErrors || reporter.WARNING.count > 0) {
+    if (reporter.hasErrors /* || reporter.hasWarnings*/ ) {
```
Comment (juanpedromoreno) [inline server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala]: What about this commented condition? I guess is still valid, isn't it?

File: server/src/main/scala/org/scalaexercises/evaluator/untitled.sc [hunk 4]
```diff
@@ -0,0 +1,13 @@
+"""
```
Comment (juanpedromoreno) [inline server/src/main/scala/org/scalaexercises/evaluator/untitled.sc]: Is this file needed?

File: .travis.yml [hunk 5]
```diff
@@ -3,7 +3,7 @@ sudo: required
 services:
 - docker
 scala:
-- 2.12.10
+- 2.13.1
```
Comment (juanpedromoreno) [inline .travis.yml]: You also need to change the conditions below: `"$TRAVIS_SCALA_VERSION" = "2.12.10"`

File: server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala [hunk 6]
```diff
@@ -96,68 +108,60 @@ class Evaluator[F[_]: Sync](timeout: FiniteDuration = 20.seconds)(
...
-      dependencies: Seq[Dependency] = Nil
+      dependencies: Seq[EvaluatorDependency] = Nil
   ): F[EvalResult[T]] = {
     for {
       allJars <- fetchArtifacts(remotes, dependencies)
       result <- allJars match {
         case Right(jars) =>
-          timeoutTo[EvalResult[T]](F.delay { evaluate(code, jars) }, timeout, Timeout[T](timeout))
+          evaluate(code, jars)
+            .timeoutTo(timeout, Timeout[T](timeout).asInstanceOf[EvalResult[T]].pure[F])
```
Comment (juanpedromoreno) [inline server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala]: I'm not pretty sure what you want to do here, shouldn't be something like this?

```scala
result <- allJars match {
        case Right(jars) =>
          evaluate[T](code, jars).timeout(timeout)
        case Left(fileError) => F.pure(UnresolvedDependency[T](fileError.describe))
      }
```

Comment (kiroco12) [reply server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala]: When the execution reaches a timeout we want to return an `EvalResult`. The `timeout` method throws an exception when the execution takes more than the specified time. Should I use `timeout` with `handleError`?

File: server/src/test/scala/org/scalaexercises/evaluator/EvalEndpointSpec.scala [hunk 7]
```diff
@@ -76,8 +79,10 @@ class EvalEndpointSpec extends AnyFunSpec with Matchers with Implicits {
           EvalRequest(
             code = "{ while(true) {}; 123 }",
             resolvers = commonResolvers,
-            dependencies = scalaDependencies(Scala212)),
-          `X-Scala-Eval-Api-Token`(validToken)),
+            dependencies = scalaDependencies(Scala212)
```
Comment (juanpedromoreno) [inline server/src/test/scala/org/scalaexercises/evaluator/EvalEndpointSpec.scala]: Can we replace the Scala 2.11 examples by Scala 2.13?

File: server/src/test/scala/org/scalaexercises/evaluator/EvalEndpointSpec.scala [hunk 8]
```diff
@@ -91,7 +96,8 @@ class EvalEndpointSpec extends AnyFunSpec with Matchers with Implicits {
             code = "{import cats._; Eval.now(42).value}",
             resolvers = commonResolvers,
             dependencies = List(Dependency("org.typelevel", "cats_2.11", "0.6.0")) ++ scalaDependencies(
```
Comment (juanpedromoreno) [inline server/src/test/scala/org/scalaexercises/evaluator/EvalEndpointSpec.scala]: We should upgrade the dependencies according to the new library stack
